mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -966,6 +966,18 @@ LogicalResult tosa::TransposeOp::inferReturnTypeComponents(
   if (matchPattern(adaptor.getPerms(), m_Constant(&attr)) &&
       attr.getType().getRank() == 1) {
     ShapeAdaptor permShape = attr;
+    // Constant permutation must be the same length as the input rank.
+    if (inputShape.getRank() != permShape.getRank())
+      return emitOptionalError(location,
+                               "Constant permutation must be the same length"
```

Comment:
Convention for errors via emitError etc is sentence fragments (start lower cased and no trailing punctuation)